src/plugins/english/mvlempyr.ts

```diff
@@ -0,0 +1,243 @@
+
+  async parseNovelListPage(
+    url: string,
+    nextPageConsumer: (nextPage: string) => void,
+  ): Promise<Plugin.NovelItem[]> {
+    const result = await fetchApi(url);
+    const body = await result.text();
+
+    const loadedCheerio = parseHTML(body);
+    return this.parseNovels(loadedCheerio, nextPageConsumer);
+  }
+
+  async parseNovel(novelPath: string): Promise<Plugin.SourceNovel> {
+    const url = this.site + novelPath;
+    const result = await fetchApi(url);
+    const body = await result.text();
+
+    const loadedCheerio = parseHTML(body);
+
+    this.checkCaptcha(loadedCheerio);
+
+    const code = loadedCheerio('#novel-code').text();
+    const tags = (
+      await fetchApi(this._chapSite + 'wp-json/wp/v2/tags?slug=' + code, {
+        headers: {
+          origin2: this.site.replace(/\/$/, ''),
+          origin: this.site.replace(/\/$/, ''),
+        },
+      }).then(res => res.json())
+    )[0];
+
+    const posts = (
+      await Promise.all(
+        new Array(Math.ceil(tags.count / 500))
+          .fill(0)
+          .map((_, i) => i + 1)
+          .map(page =>
+            fetchApi(
+              this._chapSite +
+                'wp-json/wp/v2/posts?tags=' +
+                tags.id +
+                '&per_page=500&page=' +
+                page,
+              {
+                headers: {
+                  // origin2: this.site.replace(/\/$/, ''),
+                  origin: this.site.replace(/\/$/, ''),
+                },
+              },
+            ).then(res => res.json()),
+          ),
+      )
+    )
+      .flat()
+      .sort((a, b) => a.acf.chapter_number - b.acf.chapter_number);
+
+    return {
+      path: novelPath,
+      name:
+        loadedCheerio('div.image-container.w-embed > img').attr('alt') ||
+        'Untitled',
+      cover: loadedCheerio('div.image-container.w-embed > img').attr('src'),
+      summary: loadedCheerio('div.synopsis.w-richtext').text().trim(),
+      chapters: posts.map(chap => ({
+        name: chap.acf.ch_name,
+        path: 'chapter/' + chap.acf.novel_code + '-' + chap.acf.chapter_number,
+        date: chap.date,
+        chapterNumber: chap.acf.chapter_number,
+      })),
+      status: loadedCheerio('div.novelstatustextmedium').text(),
+      author: loadedCheerio('div.mobileauthorname').text(),
+      genres: loadedCheerio('div.novelgenre.mobile > div > div > a > div')
+        .map((i, el) => loadedCheerio(el).text())
+        .toArray()
+        .join(','),
+    };
+  }
+
+  async parseChapter(chapterPath: string): Promise<string> {
+    const result = await fetchApi(this.site + chapterPath);
+    const body = await result.text();
+
+    const loadedCheerio = parseHTML(body);
+    this.checkCaptcha(loadedCheerio);
+
+    return loadedCheerio('#chapter').html() || '';
+  }
+
+  async searchNovels(
+    searchTerm: string,
+    page: number,
+  ): Promise<Plugin.NovelItem[]> {
+    if (page > 1) return [];
+
+    const allNovels = await this.getAllNovels();
```

Comment:
yes this is actually how their website does searching :skull:
(made it cache locally tho as i assume the list of novels wont be changed too often)